pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -185,6 +180,72 @@ func (r *reconciler) createBuildsForArchitectures(ctx context.Context, mabc *v1.
 	return nil
 }
 
+func (r *reconciler) handlePushImageWithManifest(mabc *v1.MultiArchBuildConfig, targetImageRef string) (func(mabcToMutate *v1.MultiArchBuildConfig), bool) {
+	if getCondition(mabc, PushImageManifestDone) != nil {
+		return nil, true
+	}
+
+	if err := r.manifestPusher.PushImageWithManifest(mabc.Status.Builds, targetImageRef); err != nil {
+		return func(mabcToMutate *v1.MultiArchBuildConfig) {
+			mabcToMutate.Status.Conditions = append(mabcToMutate.Status.Conditions, metav1.Condition{
+				Type:               PushImageManifestDone,
+				Status:             metav1.ConditionFalse,
+				LastTransitionTime: metav1.Time{Time: time.Now()},
+				Reason:             "PushManifestError",
+				Message:            err.Error(),
+			})
+			mabcToMutate.Status.State = v1.FailureState
+		}, false
+	}
+
+	return func(mabcToMutate *v1.MultiArchBuildConfig) {
+		mabcToMutate.Status.Conditions = append(mabcToMutate.Status.Conditions, metav1.Condition{
+			Type:               PushImageManifestDone,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: time.Now()},
+			Reason:             "PushManifestSuccess",
+		})
+	}, true
+}
+
+// runPostBuildActions ensure some actions are performed after the builds have been spawned.
+// Actions are performed in order, one after the other. If at least one fails then the chain breaks
+// and the overall status is set to failed.
+func (r *reconciler) runPostBuildActions(actions ...func() (func(mabcToMutate *v1.MultiArchBuildConfig), bool)) func(mabcToMutate *v1.MultiArchBuildConfig) {
```

Comment:
You shouldn't be needed to care about this. You only need to run the mirroring after the manifestPusher succeeds.